spec/user_interface_spec.rb

```diff
@@ -29,4 +29,39 @@
 
     expect(ui.user_input).to eq('1')
   end
+
+  it 'validates input is a number' do
+    input = StringIO.new('a')
+    ui = described_class.new(input, output)
+
+    ui.run
+
+    expect(output.string).to include(described_class::ERROR_MESSAGE)
+  end
+
+  it 'validates input only if the number is 1' do
+    input = StringIO.new('2')
+    ui = described_class.new(input, output)
+
+    ui.run
+
+    expect(output.string).to include(described_class::ERROR_MESSAGE)
+  end
+
+  it 'reads the input again if input is invalid' do
+    input = StringIO.new('2')
+    ui = described_class.new(input, output)
+
+    ui.run
+    input.string = '1'
+
+    expect(ui.user_input).to eq('1')
+  end
+
+  it 'returns a valid input' do
+    input = StringIO.new('1')
+    ui = described_class.new(input, output)
```

Comment:
Remember you already defined a `let` variable for a valid input, so you can use it here:
```ruby
ui = described_class.new(valid_input, output)
```